src/libraries/System.Private.CoreLib/src/System/Collections/HashHelpers.cs

```diff
@@ -101,10 +101,8 @@ public static uint FastMod(uint value, uint divisor, ulong multiplier)
             // which allows to avoid the long multiplication if the divisor is less than 2**31.
             Debug.Assert(divisor <= int.MaxValue);
 
-            ulong lowbits = multiplier * value;
-            // 64bit * 64bit => 128bit isn't currently supported by Math https://github.com/dotnet/runtime/issues/31184
-            // otherwise we'd want this to be (uint)Math.BigMul(lowbits, divisor, out _)
-            uint highbits = (uint)((((lowbits >> 32) + 1) * divisor) >> 32);
+            // This is equivalent of (uint)Math.BigMul(multiplier * value, divisor, out _) that is currently faster
```

Comment:
I'm confused by the comment - it sounds like BigMul is faster but I think you mean the code written here is faster than BigMul because it only needs the high bits?

Reply:
Comments updated.